src/controllers/authentication.controller.ts

```diff
@@ -0,0 +1,166 @@
+// Uncomment these imports to begin using these cool features!
+import {TokenService, UserService} from '@loopback/authentication';
+import {TokenServiceBindings, User, UserServiceBindings} from '@loopback/authentication-jwt';
+import {inject} from '@loopback/core';
+import {repository} from '@loopback/repository';
+import {api, getModelSchemaRef, HttpErrors, post, requestBody} from '@loopback/rest';
+import _ from 'lodash';
+import winston from 'winston';
+import {LogConfig} from '../config/logConfig';
+import {PasswordHasherBindings} from '../keys';
+import {NewUserRequest} from '../models';
+import {PasswordHasher} from '../services/hash.password.bcrypt';
+import {validateCredentials} from '../services/validator';
+import {Credentials, UserRepository} from './../repositories/user.repository';
+
+@api({basePath: '/auth'})
+export class AuthenticationController {
+  constructor(
+    @repository(UserRepository) public userRepository: UserRepository,
+    @inject(PasswordHasherBindings.PASSWORD_HASHER)
+    public passwordHasher: PasswordHasher,
+    @inject(TokenServiceBindings.TOKEN_SERVICE)
+    public jwtService: TokenService,
+    @inject(UserServiceBindings.USER_SERVICE)
+    public userService: UserService<User, Credentials>,
+    // Inject Winston logger here
+    public logger = winston.loggers.get(LogConfig.logName),
+  ) {}
+
+  /**
+   * Flow:
+   * Takes user details
+   * Extracts and hashes plain password
+   * Saves User details with hashed password
+   * Uses provided user details to generate
+   * JWT token.
+   * @param newUserRequest
+   * @returns token
+   */
+  @post('/signup', {
+    responses: {
+      '200': {
```

Comment:
there are some error cases not documented here.

Reply:
Yes, you are right! I had that in mind initially but I think it escaped my mind before submitting the PR.
Adding the following will the endpoint will help. I have added it to the code as well but because of the closed PR, changes are not reflected here. 

```js  '400': {
        description: 'Wrong Request body Structure',
        content: {
          'application/json': {
            schema: {
              type: 'object',
              properties: {
                error: {
                  type: 'object',
                  properties: {
                    statusCode: {
                      type: 'integer'
                    },
                    name: {
                      type: 'string'
                    },
                    message: {
                      type: 'string'
                    } 
                  }
                }
              }
            }
          }
        }
      },
      '422': {
        description: 'Invalid Request body',
        content: {
          'application/json': {
            schema: {
              type: 'object',
              properties: {
                error: {
                  type: 'object',
                  properties: {
                    statusCode: {
                      type: 'integer'
                    },
                    name: {
                      type: 'string'
                    },
                    message: {
                      type: 'string'
                    },
                    code: {
                      type: 'integer'
                    },
                    details: {
                      type: 'array',
                      items: {
                        type: 'object'
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }